controller-server/src/test/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializerTest.java

```diff
@@ -112,6 +112,7 @@ public void testSerialization() {
                                                Optional.of(User.from("by-username")),
                                                OptionalInt.of(7),
                                                new MetricsService.ApplicationMetrics(0.5, 0.9),
+                                               Optional.of("---begin---\nKEY\n---end---"),
```

Comment:
I don't think this is valid, it needs to be "BEGIN CERTIFICATE" and 5 dashes, not 3. :-)
Let's make sure we have some basic verification of the PEM file on upload so we can return a proper error message at that time, and not just fail on submit later.

Reply:
Since this is for a public key in pem format it should be: 
-----BEGIN PUBLIC KEY-----
KEY....
-----END PUBLIC KEY-----

Reply:
Definitely. With submissions, we should do all the verification we can before storing the package.

Reply:
Upload _is_ submission, btw :)